internal/core/service/template_service.go

```diff
@@ -0,0 +1,55 @@
+package service
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	repositories "github.com/cyneptic/letsgo-smspanel/infrastructure/repository"
+	"github.com/cyneptic/letsgo-smspanel/internal/core/entities"
+	"github.com/cyneptic/letsgo-smspanel/internal/core/ports"
+)
+
+type TemplateService struct {
+	db ports.TemplateRepositoryContract
+}
+
+func NewTemplateService() *TemplateService {
+	db := repositories.NewGormDatabase()
+	return &TemplateService{
+		db: db,
+	}
+}
+
+func (svc *TemplateService) CreateTemplate(temp entities.Template) error {
+	err := svc.db.AddTemplate(temp)
```

Comment:
I cannot find where the id for template is generated. in repository its set to temp.ID, but no temp.ID is generated via uuid.New() here.